third_party/rust/halo2_proofs/v0_3/BUILD.gn

```diff
@@ -0,0 +1,104 @@
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/plonk/lookup/verifier.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/plonk/permutation.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/plonk/permutation/keygen.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/plonk/permutation/prover.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/plonk/permutation/verifier.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/plonk/prover.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/plonk/vanishing.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/plonk/vanishing/prover.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/plonk/vanishing/verifier.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/plonk/verifier.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/plonk/verifier/batch.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/poly.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/poly/commitment.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/poly/commitment/msm.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/poly/commitment/prover.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/poly/commitment/verifier.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/poly/domain.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/poly/evaluator.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/poly/multiopen.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/poly/multiopen/prover.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/poly/multiopen/verifier.rs",
+    "//brave/third_party/rust/halo2_proofs/v0_3/crate/src/transcript.rs",
+  ]
+  inputs = []
+
+  build_native_rust_unit_tests = false
+  edition = "2021"
+  cargo_pkg_version = "0.3.0"
+  cargo_pkg_authors = "Sean Bowe <sean@electriccoin.co>, Ying Tong Lai <yingtong@electriccoin.co>, Daira Hopwood <daira@electriccoin.co>, Jack Grigg <jack@electriccoin.co>"
+  cargo_pkg_name = "halo2_proofs"
+  cargo_pkg_description =
+      "Fast PLONK-based zero-knowledge proving system with no trusted setup"
+  library_configs -= [ "//build/config/compiler:chromium_code" ]
+  library_configs += [ "//build/config/compiler:no_chromium_code" ]
+  executable_configs -= [ "//build/config/compiler:chromium_code" ]
+  executable_configs += [ "//build/config/compiler:no_chromium_code" ]
+  proc_macro_configs -= [ "//build/config/compiler:chromium_code" ]
+  proc_macro_configs += [ "//build/config/compiler:no_chromium_code" ]
+  deps = [
+    "//brave/third_party/rust/blake2b_simd/v1:lib",
+    "//brave/third_party/rust/ff/v0_13:lib",
+    "//brave/third_party/rust/group/v0_13:lib",
+    "//brave/third_party/rust/halo2_legacy_pdqsort/v0_1:lib",
+    "//brave/third_party/rust/maybe_rayon/v0_1:lib",
```

Comment:
Seems like we're not enabling the multicore feature, so we don't need `maybe_rayon`?

Reply:
I believe maybe_rayon is needed to decide whether to use rayon or not.

Reply:
Ah, sorry. rayon is optional in maybe_rayon, but halo2_proofs uses the maybe_rayon polyfill even in the serial case, which is only about a hundred lines. As long as we don't enabled the `multicore` feature on halo2, which enables the `threads` feature on maybe_rayon, we don't import rayon as a transitive dependency, and the thread-api calls are all implemented by serial stubs. So this is Ok.